substrate/frame/support/src/traits/tokens/fungible/mod.rs

```diff
@@ -17,26 +17,129 @@
 //!   funds must be removed from an account before it is known precisely what should be done with
 //!   them.
+//!
+//! ## Terminology
+//!
+//! - **Total Issuance**: The total number of units in existence in a system.
+//!
+//! - **Total Balance**: The sum of an account's free and held balances.
+//!
+//! - **Free Balance**: A portion of an account's total balance that is not held.
+//!
+//! - **Held Balance**: Held balance still belongs to the account holder, but is suspended. It can
+//!   be slashed, but only after all the free balance has been slashed.
+//!
+//!   Multiple holds stack rather than overlay. This means that if an account has
+//!   3 locks for 100 units, the account can spend its funds for any reason down to 300 units, at
+//!   which point the holds will start to come into play.
+//!
+//! - **Frozen Balance**: A freeze on a specified amount of an account's free balance until a
+//!   specified block number.
+//!
+//!   Multiple freezes always operate over the same funds, so they "overlay" rather than
+//!   "stack". This means that if an account has 3 freezes for 100 units, the account can spend its
+//!   funds for any reason down to 100 units, at which point the freezes will start to come into
+//!   play.
+//!
+//! - **Minimum Balance (a.k.a. Existential Deposit, a.k.a. ED)**: The minimum balance required to
+//!   create or keep an account open. This is to prevent "dust accounts" from filling storage. When
+//!   the free plus the held balance (i.e. the total balance) fall below this, then the account is
+//!   said to be dead; and it loses its functionality as well as any prior history and all
+//!   information on it is removed from the chain's state. No account should ever have a total
+//!   balance that is strictly between 0 and the existential deposit (exclusive). If this ever
+//!   happens, it indicates either a bug in the implementation of this trait or an erroneous raw
+//!   mutation of storage.
+//!
+//! - **Untouchable Balance**: The part of a user's free balance they cannot spend, due to ED or
+//!   Freeze(s).
+//!
+//! - **Spendable Balance**: The part of a user's free balance they can spend.
+//!
+//! - **Imbalance**: A condition when some funds were credited or debited without equal and opposite
+//!   accounting (i.e. a difference between total issuance and account balances). Functions that
+//!   result in an imbalance will return an object of the [`imbalance::Credit`] or
+//!   [`imbalance::Debt`] traits that can be managed within your runtime logic.
+//!
+//!   If an imbalance is simply dropped, it should automatically maintain any book-keeping such as
+//!   total issuance.
+//!
+//! ## Visualising Balance Components Together 💫
+//!
+//! ```ignore
+//! |__total__________________________________|
+//! |__on_hold__|_____________free____________|
+//! |__________frozen___________|
+//! |__on_hold__|__ed__|
+//!             |__untouchable__|__spendable__|
+//! ```
+//!
+//! ## Holds and Freezes
```

Comment:
Note I opted to describe Holds and Freezes together as I feel they were designed to be used together, and so make the most sense when described together.

Happy to adjust though.